INTMDB-378: Add link for How To Guide for existing container ID (#883)
mongodb/terraform-provider-mongodbatlas: website/docs/guides/howto-guide.html.markdown

---
layout: "mongodbatlas"

> page_title: "MongoDB Atlas Provider How to Guide"

Reviewer: ```suggestion
page_title: "MongoDB Atlas Provider How-To Guide"
```

sidebar_current: "docs-mongodbatlas-guides-how-to-guide"
description: |-

> MongoDB Atlas Provider : How to Guide

Reviewer: ```suggestion
MongoDB Atlas Provider : How-To Guide
```

---

> # MongoDB Atlas Provider: How to Guide

Reviewer: ```suggestion
# MongoDB Atlas Provider: How-To Guide
```

The Terraform MongoDB Atlas Provider guide to perform common tasks with the provider.

> Document for users how to get a pre-existing container id

Reviewer: ```suggestion
##How to Get A Pre-existing Container ID
```

> 

Reviewer: ```suggestion
The following is an end to end example of how to get an existing container id.  

```

> 1) Empty project

Reviewer: ```suggestion
1) Start with an empty project
```

> 2) empty state file

Reviewer: ```suggestion
2) Empty state file
```

> 3) applied curl to build cluster

Reviewer: ```suggestion
3) Apply a curl command to build cluster
```

> 4) terraform apply to retrieve container id

Reviewer: ```suggestion
4) Run `terraform apply` to retrieve the container id
```

> Validated by testing following scenario container_id populated in state after first apply ...

Reviewer: ```suggestion
The following illustrates step 3 and 4 above, assuming 1 & 2 were true: 
```

> 1) create cluster using curl command to simulate non terraform based cluster

Reviewer: ```suggestion
1) Create a cluster using a curl command to simulate non-Terraform created cluster.  This will also create a container.  
```

```
curl --user "pub:priv" --digest \
--header "Content-Type: application/json" \
--include \
--request POST "https://cloud.mongodb.com/api/atlas/v1.0/groups/grpid/clusters?pretty=true" \
--data '
{
  "name": "SingleRegionCluster",

>   "diskSizeGB": 100,

Reviewer: Trying to get users to get out of the habit of using this one unless the want more than the default storage so let's remove it as it should be optional.  Though we should test that as the docs makes it sound like it is required but it shouldn't be. 
```suggestion
```

  "numShards": 1,
  "providerSettings": {
    "providerName": "AWS",
    "instanceSizeName": "M40",
    "regionName": "US_EAST_1"
  },
  "clusterType": "REPLICASET",
  "replicationFactor": 3,
  "replicationSpecs": [
    {
      "numShards": 1,
      "regionsConfig": {
        "US_EAST_1": {
          "analyticsNodes": 0,
          "electableNodes": 3,
          "priority": 7,
          "readOnlyNodes": 0
        }
      },
      "zoneName": "Zone 1"
    }
  ],
  "backupEnabled": false,

>   "providerBackupEnabled": true,

Reviewer: This is old, we shouldn't use it imo 
```suggestion
```

  "autoScaling": {
    "diskGBEnabled": true
  }
}'
```

 

> 2) Apply this terraform to read information from datasources

Reviewer: ```suggestion
2) The apply this Terraform config to then read the information from the appropriate Data Sources and output the container id.  
```

 
```
data "mongodbatlas_cluster" "admin" { 
  name = "SingleRegionCluster" 
  project_id = local.mongodbatlas_project_id 
}

data "mongodbatlas_network_container" "admin" { 
  project_id = local.mongodbatlas_project_id 
  container_id = data.mongodbatlas_cluster.admin.container_id 
}

output "container" { 
  value = data.mongodbatlas_network_container.admin.container_id 
}

Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

container = "62ffe4ecb79e2e007c375935"
```
 

> Using versions

Reviewer: ```suggestion
This example was tested using versions:
```
